commit 2014face1e99077aa7253f62615454f402aef002
Author: Phil Elwell <phil@raspberrypi.org>
Date:   Thu Jun 25 12:16:11 2015 +0100

    gpio-poweroff: Allow it to work on Raspberry Pi
    
    The Raspberry Pi firmware manages the power-down and reboot
    process. To do this it installs a pm_power_off handler, causing
    the gpio-poweroff module to abort the probe function.
    
    This patch introduces a "force" DT property that overrides that
    behaviour, and also adds a DT overlay to enable and control it.
    
    Note that running in an active-low configuration (DT parameter
    "active_low") requires a custom dt-blob.bin and probably won't
    allow a reboot without switching off, so an external inversion
    of the trigger signal may be preferable.

diff --git a/drivers/power/reset/gpio-poweroff.c b/drivers/power/reset/gpio-poweroff.c
index be3d81ff51cc..a030ae9fb1fc 100644
--- a/drivers/power/reset/gpio-poweroff.c
+++ b/drivers/power/reset/gpio-poweroff.c
@@ -49,9 +49,11 @@ static int gpio_poweroff_probe(struct platform_device *pdev)
 {
 	bool input = false;
 	enum gpiod_flags flags;
+	bool force = false;
 
 	/* If a pm_power_off function has already been added, leave it alone */
-	if (pm_power_off != NULL) {
+	force = of_property_read_bool(pdev->dev.of_node, "force");
+	if (!force && (pm_power_off != NULL)) {
 		dev_err(&pdev->dev,
 			"%s: pm_power_off function already registered",
 		       __func__);
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1061_2014face1e99077aa7253f62615454f402aef002.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1061_2014face1e99077aa7253f62615454f402aef002.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
